src/main/java/net/fabricmc/loader/EntrypointStorage.java

```diff
@@ -101,23 +101,21 @@ public String toString() {
 		@SuppressWarnings("unchecked")
 		@Override
 		public <T> T getOrCreate(Class<T> type) throws Exception {
-			Object o = instanceMap.get(type);
-
-			if (o == null) {
-				o = create(type);
-				instanceMap.put(type, o);
-			}
-
-			return (T) o;
+			//noinspection unchecked
+			return (T) instanceMap.computeIfAbsent(type, this::create);
```

Comment:
Is it worth doing this if it's not actually any shorter?

Reply:
Why not, this saves a map lookup, and the entrypoint exception is just thrown as usual

Reply:
`IdentityHashMap#computeIfAbsent` uses the default implementation on `Map` though so the code is the same in terms of map lookups, it's just more indirect. It doesn't entirely make sense that `create` was ever a separate method really when this all could just be
```java
@Override
@SuppressWarnings("unchecked")
public <T> T getOrCreate(Class<T> type) throws Exception {
	Object o = instanceMap.get(type);

	if (o == null) {
		instanceMap.put(type, o = adapter.create(mod, value, type));
	}

	return (T) o; //Or even return type.cast(o) if you really wanted to be type safe/paranoid
}
```

Reply:
Well, how can you rule out that in a future jdk they will optimize `computeIfAbsent` for identity hash maps? If they do, this current change can benefit from that.

Reply:
I looked at the code around. I find no reason to use identity hash mas, especially given that classes are already identity-based objects. I am planning to use hash maps with capacity 1 and create an adaption result class (to hold the exception so we don't reevaluate exceptions, I assume)